src/circular_buffer.jl

```diff
@@ -72,12 +72,14 @@ function Compat.popfirst!(cb::CircularBuffer)
 end
 
 function Compat.pushfirst!(cb::CircularBuffer, data)
-    # if full, decrement and overwrite, otherwise pushfirst
-    if cb.length < cb.capacity
+    # if full, decrement and overwrite, otherwise unshift
+    if length(cb) == cb.capacity
+        cb.first = (cb.first == 1 ? cb.capacity : cb.first - 1)
+        cb[1] = data
+    else
         cb.length += 1
+        unshift!(cb.buffer, data)
```

Comment:
For 0.7 reasons don't use `unshift!`, use `pushfirst!`  inside the buffer as well

Reply:
Oops, thanks.